core/src/snapshot/postprocessers/postprocess.rs

```diff
@@ -0,0 +1,198 @@
+        };
+
+        println!("logical_expression: {}",  logical_expression); 
+
+        println!("checking if parenthesis necessary");
+        // check if the parentheses are unnecessary and remove them if so
+        if are_parentheses_unnecessary(&logical_expression) {
+            println!("parenthesis are unnecessary");
+            cleaned.replace_range(
+                paren_start..paren_end,
+                match logical_expression.get(2..logical_expression.len() - 2) {
+                    Some(x) => x,
+                    None => "",
+                },
+            );
+            println!("cleaned after replace_range {}", cleaned);
+
+
+            // remove double negation, if one was created
+            if cleaned.contains("!!") {
+                println!("cleared double negation");
+                cleaned = cleaned.replace("!!", "");
+            }
+
+            // recurse into the next set of parentheses
+            // don't increment the paren_index because we just removed a set
+            cleaned = simplify_parentheses(&cleaned, paren_index);
+        } else {
+            // remove double negation, if one exists
+            if cleaned.contains("!!") {
+                println!("cleared double negation");
+                cleaned = cleaned.replace("!!", "");
+            }
+
+            // recurse into the next set of parentheses
+            println!("parenthesis are necessary - recursing");
+            cleaned = simplify_parentheses(&cleaned, paren_index + 1);
+        }
+    }
+
+    cleaned
+}
+
+pub fn cleanup(line: &str) -> String {
+    let line = simplify_parentheses(line, 0);
+    println!("line before removing double negation: {}", line);
+
+    remove_double_negation(&line)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_simplify_parentheses() {
+        let line = "((arg0))";
+        assert_eq!(simplify_parentheses(line, 0), "arg0");
+    }
+
+    #[test]
+    fn test_simplify_parentheses_complex() {
+        let line = "if ((cast(((arg0) + 1) / 10))) {";
+        assert_eq!(simplify_parentheses(line, 0), "if (cast((arg0 + 1) / 10)) {");
+    }
+
+    #[test]
+    fn test_simplify_parentheses_complex2() {
+        let line = "if (((((((((((((((cast(((((((((((arg0 * (((((arg1))))))))))))) + 1)) / 10)))))))))))))))) {";
+        assert_eq!(simplify_parentheses(line, 0), "if (cast(((arg0 * (arg1)) + 1) / 10)) {");
+    }
+
+    #[test]
+    fn test_remove_double_negation_and_simplify_parenthesis() {
+        let line = "if (!(storage [0x08] > (storage [0x08] + ((!((!((argO * storage [0x08]))))) * ( ( ( (arg® * storage [0x08]) - 0x01) / storage [0x021) + 0×01)))))) { .. }";
+        let expected = "if (!storage [0x08] > (storage [0x08] + ((argO * storage [0x08]) * ( ( ( (arg® * storage [0x08]) - 0x01) / storage [0x021) + 0×01))))) { .. }";
+        assert_eq!(simplify_parentheses(line, 0), expected);
+    }
+    
+    #[test]
+    fn test_double_negation() {
```

Comment:
Is this test accurate?

Reply:
this is where I was confused -- in fact we should expect:
`if (storage[keccak256(memory[0])] == Oxfffffffffffffffffffffffffffffffffffffffff)`

Reply:
yep that makes sense to me

Reply:
really struggling on these nested negations like !(!(...)), maybe I could put up another issue in case someone wants to help?

Reply:
i can help out with this!